.github/workflows/sonar.yml

```diff
@@ -60,14 +59,20 @@ jobs:
           key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
           restore-keys: ${{ runner.os }}-m2
       - name: SonarCloud Scan on PR
-        if: github.event.workflow_run.event == 'pull_request'
         run: mvn -B clean verify -Pci org.sonarsource.scanner.maven:sonar-maven-plugin:3.9.1.2184:sonar -Dsonar.projectKey=kroxylicious_kroxylicious -Dsonar.scm.revision=${{ github.event.workflow_run.head_sha }} -Dsonar.pullrequest.key=${{ fromJson(steps.get_pr_data.outputs.data).number }} -Dsonar.pullrequest.branch=${{ fromJson(steps.get_pr_data.outputs.data).head.ref }} -Dsonar.pullrequest.base=${{ fromJson(steps.get_pr_data.outputs.data).base.ref }}
         env:
           SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      - name: SonarCloud Scan on push
-        if: github.event.workflow_run.event == 'push' && github.event.workflow_run.head_repository.full_name == github.event.repository.full_name
-        run: mvn -B clean verify -Pci org.sonarsource.scanner.maven:sonar-maven-plugin:3.9.1.2184:sonar -Dsonar.scm.revision=${{ github.event.workflow_run.head_sha }} -Dsonar.branch.name=${{ github.event.workflow_run.head_branch }}
+  sonarcloud:
+    name: SonarCloud
+    runs-on: ubuntu-latest
+    if: github.event.workflow_run.event == 'push' && github.event.github.ref_name == main
```

Comment:
do we need the redundant ref_name check since main is specified in the `on`?

Reply:
🤷‍♂️

Reply:
Is it redundant? I think the action will be triggered if the build workflow is run anywhere and those will be triggered by a push so I think this ensures that we are only trying this on main but I could easily be wrong

Reply:
Just confusing myself. What does the:

```
on:
    push:
        branches:
            - main
```

trigger?

Reply:
I was wondering if `github.event.workflow_run` is only present when triggered by the other part of the `on`.

Reply:
I think the on conditions are an `or` so `push.branches.[main]` only triggers on pushes to main as you would expect but it could also be triggered by the build workflow completing.

Reply:
so `github.event.workflow_run` is present in the push event too?

Reply:
I think so as its the metadata about the current run